code/lib/cli/src/generators/REACT/index.ts

```diff
@@ -1,8 +1,19 @@
+import { detectLanguage } from '../../detect';
+import { SupportedLanguage } from '../../project_types';
 import { baseGenerator } from '../baseGenerator';
 import type { Generator } from '../types';
 
 const generator: Generator = async (packageManager, npmOptions, options) => {
-  await baseGenerator(packageManager, npmOptions, options, 'react');
+  // Add prop-types dependency if not using TypeScript
+  const extraPackages = [];
+  const language = detectLanguage();
+  if (language === SupportedLanguage.JAVASCRIPT) {
+    extraPackages.push('prop-types');
+  }
```

Comment:
I might write this as:
```suggestion
  const language = detectLanguage();
  const extraPackages = language === SupportedLanguage.JAVASCRIPT ? ['prop-types'] : [];
```